aspnetcore/blazor/hybrid/security/index.md

```diff
@@ -5,23 +5,580 @@ description: Learn about Blazor Hybrid authentication and authorization scenario
-ms.date: 04/19/2022
+ms.date: 05/12/2022
 no-loc: [".NET MAUI", "Mac Catalyst", "Blazor Hybrid", Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
 uid: blazor/hybrid/security/index
+zone_pivot_groups: blazor-hybrid-frameworks
 ---
 # ASP.NET Core Blazor Hybrid authentication and authorization
 
 This article describes ASP.NET Core's support for the configuration and management of security in Blazor Hybrid apps.
 
 [!INCLUDE[](~/blazor/includes/blazor-hybrid-preview-notice.md)]
 
+## Overview
+
+Blazor Hybrid apps that render web content execute .NET code inside a platform :::no-loc text="Web View":::. The .NET code interacts with the web content via an interop channel between the two.
+
+![The WebView and .NET code interoperate within the app to render web content.](~/blazor/hybrid/security/index/_static/figure01.png)
+
+The web content rendered into the :::no-loc text="Web View"::: can come from assets provided by the app from either of the following locations:
+
+* The `wwwroot` folder in the app.
+* A source external to the app. For example, a network source, such as the Internet.
+
+A trust boundary exists between the .NET code and the code that runs inside the :::no-loc text="Web View":::. .NET code is provided by the app and any trusted third-party packages that you've installed. After the app is built, the .NET code :::no-loc text="Web View"::: content sources can't change.
+
+In contrast to the .NET code sources of content, content sources from the code that runs inside the :::no-loc text="Web View"::: can come not only from the app but also from external sources. For example, static assets from an external Content Delivery Network (CDN) might be used or rendered by an app's :::no-loc text="Web View":::.
+
+Consider the code inside the :::no-loc text="Web View"::: as **untrusted** in the same way that code running inside the browser for a web app isn't trusted. The same threats and general security recommendations apply to untrusted resources in Blazor Hybrid apps as for other types of apps. If possible, avoid loading content from a third-party origin. To mitigate risk, you might be able to serve content directly from the app by downloading the external assets, verifying that they're safe to serve to users, and placing them into the app's `wwwroot` folder for packaging with the rest of the app.
+
+If your app must reference content from an external origin, we recommended that you use common web security approaches to provide the app with an opportunity to block the content from loading if the content is compromised:
+
+* Serve content securely with TLS/HTTPS.
+* Institute a [Content Security Policy (CSP)](https://developer.mozilla.org/docs/Web/HTTP/CSP).
+* Perform [subresource integrity](https://developer.mozilla.org/docs/Web/Security/Subresource_Integrity) checks.
+* Use an anti-virus/anti-malware scanner.
```

Comment:
This last point is a bit uncalled for. An anti-virus will not do anything in this context.

Reply:
Ok ... I wasn't aware if a service could be used in this context.

![mccoy](https://user-images.githubusercontent.com/1622880/168468190-ca8f9616-08fb-42ad-b730-dd48789c86bc.png)

### *Damn it Jim! I'm a web dev, not a desktop/mobile guy!*

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&mdash; Dr. Leonard H. "Bones" McCoy ([DeForest Kelley](https://en.wikipedia.org/wiki/DeForest_Kelley)) [Star Trek](http://www.startrek.com/), ©1966-69 [Paramount Pictures](http://www.paramount.com)